File: uid_report/uid_report.py
#!/usr/bin/env python

# Written by: Matt Hersant

import subprocess

lines = subprocess.check_output(['getent', 'passwd']).splitlines()

mystruct = {}
for line in lines:
    items = line.split(':')
    mystruct[int(items[2])] = items

for uid in sorted(mystruct.keys()):
    if uid >= 100:
        print ':'.join(mystruct[uid])

File: uid_report/uid_report.pl
#!/usr/bin/env perl

# Written by: Matt Hersant

use strict;
use Data::Dumper;

my @lines = `getent passwd`;

my %struct;
for my $line (@lines) {
    my @items = split /:/, $line;
    my $uid = $items[2];
    $struct{$uid} = \@items;
}

for my $uid (sort keys %struct) {
    if ($uid >= 100) {
        print join(':', @{$struct{$uid}});
    }
}
